pkg/api/validation/validation.go

```diff
@@ -234,6 +235,14 @@ func ValidatePositiveQuantity(value resource.Quantity, fieldName string) errs.Va
 	return allErrs
 }
 
+func ValidateImmutableField(old, new interface{}, fieldName string) errs.ValidationErrorList {
```

Comment:
Since this can only return a single error, please have it just return `error` not `errs.ValidationErrorList`